wp-includes/class-wp-service-workers.php

```diff
@@ -51,10 +93,34 @@ class WP_Service_Workers extends WP_Scripts {
 	 */
 	public $output = '';
 
+	/**
+	 * Registered caching routes and scripts.
+	 *
+	 * @var array
+	 */
+	public $registered_caching_routes = array();
+
 	/**
 	 * Initialize the class.
 	 */
 	public function init() {
+
+		if ( ! function_exists( 'list_files' ) ) {
```

Comment:
The `list_files()` function is slow. What about instead just iterating over of the registered scripts and styles in `wp_scripts()` and `wp_styles()`? Use the `ver` with each registered asset as the revision.

Reply:
Changed the logic to use `wp_scripts()` and `wp_styles()`, left in the `list_files()` for images for now within 024eab3.